### ISSUE TITLE
"keep_ckpts" not working

### ISSUE THREAD

**Issue by ThrowawayAccount01** [34j/so-vits-svc-fork#65]
In the config file, the default `"keep_ckpts": 3` does not work. When training, all checkpoints will be saved. Changing the value to any other number will not work as well.

**Comment by 34j** [comment]
I noticed it earlier too. I am having difficulty correcting it.

**Comment by ThrowawayAccount01** [comment]
I just updated to 1.2.8 but the bug is still present. All the checkpoints are still saved.

**Comment by 34j** [comment]
😭

**Comment by Lordmau5** [comment]
After a bit of back and forth with ChatGPT and looking at the documentation for `itertools.groupby` it is an issue with how the models are sorted.

The resulting array of `models_sorted` is sorted by the suffix of the G_ or D_ files, but the resulting array alternates between a "G" path and a "D" path.
This, ultimately, means that it will create a group **for every G_ and D_ path**
![image](https://user-images.githubusercontent.com/1345036/227338841-4d21b6c3-e398-4504-b023-7386fb331d72.png)

I fixed this by modifying the `sorted` method call.

Additionally, I've cleaned up the code a bit more.
Will do a pull request shortly!
